### ISSUE TITLE
AcquireTokenByIntegratedWindowsAuth[Bug] 

### ISSUE THREAD

**Issue by rajanaanraja92** [AzureAD/microsoft-authentication-library-for-dotnet#3555]
**Logs and network traces**
Exception Thrown: System.Net.Sockets.SocketException in System.dll

**Which version of MSAL.NET are you using?**
MSAL.NET 4.30.0

**Platform**
.NET 4.7.2

**What authentication flow has the issue?**
    * [ ] Integrated Windows Authentication
 
* Web app 

    * [ ] On-Behalf-Of
* Daemon app 
    * [ ] Service to Service calls

Other?

**Is this a new or existing app?**
New Web app

c. This is a new app or experiment.
-->

**Repro**

IPublicClientApplication app = PublicClientApplicationBuilder
      .Create(clientId)
      .WithAuthority(authority)
      .Build();

 var accounts = await app.GetAccountsAsync();

 AuthenticationResult result = null;
 if (accounts.Any())
 {
  result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
      .ExecuteAsync();
 }
 else
 {
  try
  {
   result = await app.AcquireTokenByIntegratedWindowsAuth(scopes)
      .ExecuteAsync(CancellationToken.None);
  }

**Expected behavior**
Get tokens and Authentication result.

**Actual behavior**
exception is thrown :  System.Net.Sockets.SocketException in System.dll

**Possible solution**

**Additional context / logs / screenshots / links to code**

Add any other context about the problem here, such as logs and screenshots, and even links to code.

**Comment by MichelZ** [comment]
Any chance you can try this with the latest version?

**Comment by rajanaanraja92** [comment]
Which latest version you are mentioning. 

I am trying to acquire token for sending email, Once it come to the below line , it is keep page loading and throwing socket exception.

result = await app.AcquireTokenByIntegratedWindowsAuth(scopes)
.ExecuteAsync(CancellationToken.None);

**Comment by MichelZ** [comment]
> Which version of MSAL.NET are you using?
MSAL.NET 4.30.0

This version is quite old. The current MSAL version is 4.45.0   (https://www.nuget.org/packages/Microsoft.Identity.Client/#versions-body-tab)

**Comment by rajanaanraja92** [comment]
I have updated my version to 4.45 through nuget package manager. But still getting the same exception.

**Comment by MichelZ** [comment]
> I have updated my version to 4.45 through nuget package manager. But still getting the same exception.

Cool. Would you mind posting the full Exception then?

**Comment by bgavrilMS** [comment]
Apart from full exception trace - did this ever work for you? Socket exception is an HTTP exception, indicating that MSAL cannot reach some authority endpoint. 

A few things that can go wrong: 
- are you behind a proxy? see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/httpclient
- see troubleshooting for this flow https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Integrated-Windows-Authentication

**Comment by rajanaanraja92** [comment]
'iisexpress.exe' (CLR v4.0.30319: /LM/W3SVC/2/ROOT-1-133040881011890959): Loaded 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\PrivateAssemblies\Runtime\Microsoft.VisualStudio.Debugger.Runtime.dll'. 
Step into: Stepping over property 'System.Threading.CancellationToken.get_None'. To step into properties or operators, go to Tools->Options->Debugging and uncheck 'Step over properties and operators (Managed only)'.
Step into: Stepping over property 'System.Runtime.CompilerServices.TaskAwaiter.get_IsCompleted'. To step into properties or operators, go to Tools->Options->Debugging and uncheck 'Step over properties and operators (Managed only)'.
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
The thread 0x25fc has exited with code 0 (0x0).
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
The thread 0x3dcc has exited with code 0 (0x0).
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
Exception thrown: 'System.Net.WebException' in System.dll
Exception thrown: 'System.Net.WebException' in System.dll
Exception thrown: 'System.ObjectDisposedException' in mscorlib.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.InvalidOperationException' in System.dll
Exception thrown: 'System.InvalidOperationException' in System.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
The thread 0x70f0 has exited with code 0 (0x0).
The thread 0x3ea0 has exited with code 0 (0x0).
The thread 0x1b48 has exited with code 0 (0x0).
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
The thread 0x33dc has exited with code 0 (0x0).
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll
Exception thrown: 'System.Net.WebException' in System.dll
Exception thrown: 'System.Net.WebException' in System.dll
Exception thrown: 'System.ObjectDisposedException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.InvalidOperationException' in System.dll
Exception thrown: 'System.InvalidOperationException' in System.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in Microsoft.Identity.Client.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in Microsoft.Identity.Client.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
Exception thrown: 'Microsoft.Identity.Client.MsalServiceException' in mscorlib.dll
The thread 0x74c8 has exited with code 0 (0x0).
Exception thrown: 'System.Net.Sockets.SocketException' in System.dll

**Comment by rajanaanraja92** [comment]
Able to access Token from Windows application through application .exe but failing for web application through IISExpress .exe.

**Comment by bgavrilMS** [comment]
In a web app or web service, you are expected to use ConfidentialClient, not PubliClient. We have not tested this flow in IIS, as we do not expect it to be used there.

**Comment by jmprieur** [comment]
it's not supposed to work on IIS, indeed.

**Comment by rajanaanraja92** [comment]
When I tried to send email from console app which is configured inside server "Services" , while  sending and save copy it is taking service account mail box instead of particular from email box mentioned in app.config .
I have configured the test shared email box in app.config file but while sending it is required another email id or service account.  Working as "Send on behalf of".

message.SendAndSaveCopy();

Our requirement is what ever mail box mentioned in app.config file which should be used as "from address" and should save emails in "Sent" folder.
